Extending RosieApplication

The documentation states "If you do not want to use Dependency Injection in your project, you do not need to extend from RosieApplication", but I get a ClassCastException when I don't extend RosieApplication: 
```
Caused by: java.lang.ClassCastException: android.app.Application cannot be cast to com.karumi.rosie.application.RosieApplication
   at com.karumi.rosie.view.RosieAppCompatActivity.injectActivityModules(RosieAppCompatActivity.java:133)
   at com.karumi.rosie.view.RosieAppCompatActivity.onCreate(RosieAppCompatActivity.java:46)
   at me.sargunvohra.android.diningcourts.MainActivity.onCreate(MainActivity.kt:17)
   at android.app.Activity.performCreate(Activity.java:6876)
   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1135)
   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3207)
   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3350) 
   at android.app.ActivityThread.access$1100(ActivityThread.java:222) 
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1795) 
   at android.os.Handler.dispatchMessage(Handler.java:102) 
   at android.os.Looper.loop(Looper.java:158) 
   at android.app.ActivityThread.main(ActivityThread.java:7229) 
   at java.lang.reflect.Method.invoke(Native Method) 
   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230) 
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120) 
```

--- a/rosie/src/main/java/com/karumi/rosie/view/RosieActivity.java
+++ b/rosie/src/main/java/com/karumi/rosie/view/RosieActivity.java
@@ -130,6 +130,9 @@ public abstract class RosieActivity extends FragmentActivity
   }
 
   private void injectActivityModules() {
+    if (!(getApplication() instanceof RosieApplication)) {
+      return;
+    }
     RosieApplication rosieApplication = (RosieApplication) getApplication();
     List<Object> additionalModules = getActivityScopeModules();
     if (additionalModules == null) {

--- a/rosie/src/main/java/com/karumi/rosie/view/RosieAppCompatActivity.java
+++ b/rosie/src/main/java/com/karumi/rosie/view/RosieAppCompatActivity.java
@@ -130,6 +130,9 @@ public abstract class RosieAppCompatActivity extends AppCompatActivity
   }
 
   private void injectActivityModules() {
+    if (!(getApplication() instanceof RosieApplication)) {
+      return;
+    }
     RosieApplication rosieApplication = (RosieApplication) getApplication();
     List<Object> additionalModules = getActivityScopeModules();
     if (additionalModules == null) {

